### DISCUSSION TITLE
v0.6.0

### DISCUSSION THREAD

**Discussion by problame** [zrepl/zrepl#648] [Announcements]
# Changelog

The original announcement is [located here](https://zrepl.github.io/v0.6.0/changelog.html).

<ul class="simple">
    <li>[FEATURE] <a class="reference internal" href="https://zrepl.github.io/v0.6.0/configuration/snapshotting.html#job-snapshotting-cron"><span class="std std-ref">Schedule-based snapshotting</span></a> using <code class="docutils literal notranslate"><span class="pre">cron</span></code> syntax instead of an interval.</li>
    <li>[FEATURE] Configurable initial replication policy.
    When a filesystem is first replicated to a receiver, this control whether just the newest
    snapshot will be replicated vs. all existing snapshots. Learn more <a class="reference internal" href="https://zrepl.github.io/v0.6.0/configuration/conflict_resolution.html#conflict-resolution-initial-replication"><span class="std std-ref">in the docs</span></a>.</li>
    <li>[FEATURE] Configurable timestamp format for snapshot names via <a class="reference internal" href="https://zrepl.github.io/v0.6.0/configuration/snapshotting.html#job-snapshotting-timestamp-format"><span class="std std-ref">timestamp_format</span></a>
    (Thanks, <a class="reference external" href="https://github.com/ydylla">@ydylla</a>).</li>
    <li>[FEATURE] Add <code class="docutils literal notranslate"><span class="pre">ZREPL_DESTROY_MAX_BATCH_SIZE</span></code> env var (default 0=unlimited)
    (Thanks, <a class="reference external" href="https://github.com/3nprob">@3nprob</a>).</li>
    <li>[FEATURE] Add <code class="docutils literal notranslate"><span class="pre">zrepl</span> <span class="pre">configcheck</span> <span class="pre">--skip-cert-check</span></code> flag (Thanks, <a class="reference external" href="https://github.com/cole-h">@cole-h</a>).</li>
    <li>[BUG] Fix resuming from interrupted replications that use <code class="docutils literal notranslate"><span class="pre">send.raw</span></code> on unencrypted datasets.<ul>
    <li>The send options introduced in zrepl 0.4 allow users to specify additional zfs send flags for zrepl to use.
    Before this fix, when setting <code class="docutils literal notranslate"><span class="pre">send.raw=true</span></code> on a job that replicates unencrypted datasets,
    zrepl would not allow an interrupted replication to resume.
    The reason were overly cautious checks to support the <code class="docutils literal notranslate"><span class="pre">send.encrypted</span></code> option.</li>
    <li>This bugfix removes these checks from the replication planner.
    This makes <code class="docutils literal notranslate"><span class="pre">send.encrypted</span></code> a sender-side-only concern, much like all other <code class="docutils literal notranslate"><span class="pre">send.*</span></code> flags.</li>
    <li>However, this means that the <code class="docutils literal notranslate"><span class="pre">zrepl</span> <span class="pre">status</span></code> UI no longer indicates whether a replication step uses encrypted sends or not.
    The setting is still effective though.</li>
    </ul>
    </li>
    <li><strong>[BREAK]</strong> convert Prometheus metric <code class="docutils literal notranslate"><span class="pre">zrepl_version_daemon</span></code> to <code class="docutils literal notranslate"><span class="pre">zrepl_start_time</span></code> metric<ul>
    <li>The metric still reports the zrepl version in a label.
    But the metric <em>value</em> is now the Unix timestamp at the time the daemon was started.
    The Grafana dashboard in <a class="reference external" href="https://github.com/zrepl/zrepl/blob/master/dist/grafana">dist/grafana</a> has been updated.</li>
    </ul>
    </li>
    <li>[BUG] transient zrepl status error: <code class="docutils literal notranslate"><span class="pre">Post</span> <span class="pre">"http://unix/status":</span> <span class="pre">EOF</span></code></li>
    <li>[BUG] don’t treat receive-side bookmarks as a replication conflict.
    This facilitates chaining of replication jobs. See <a class="reference external" href="https://github.com/zrepl/zrepl/issues/490">issue #490</a>.</li>
    <li>[BUG] workaround for Go/gRPC problem on Illumos where zrepl would
    crash when using the <code class="docutils literal notranslate"><span class="pre">local</span></code> transport type (<a class="reference external" href="https://github.com/zrepl/zrepl/issues/598">issue #598</a>).</li>
    <li>[BUG] fix active child tasks panic that cold occur during replication plannig (<a class="reference external" href="https://github.com/zrepl/zrepl/issues/193abbe">issue #193abbe</a>)</li>
    <li>[BUG] <code class="docutils literal notranslate"><span class="pre">zrepl</span> <span class="pre">status</span></code> off-by-one error in display of completed step count (<a class="reference external" href="https://github.com/zrepl/zrepl/commit/ce6701f">commit ce6701f</a>)</li>
    <li>[BUG] Allow using day &amp; week units for <code class="docutils literal notranslate"><span class="pre">snapshotting.interval</span></code> (<a class="reference external" href="https://github.com/zrepl/zrepl/commit/ffb1d89">commit ffb1d89</a>)</li>
    <li>[DOCS] <code class="docutils literal notranslate"><span class="pre">docs/overview</span></code> improvements (Thanks, <a class="reference external" href="https://github.com/jtagcat">@jtagcat</a>).</li>
    <li>[MAINT] Update to Go 1.19.</li>
    </ul>
    

## Breaking Changes

* The RPC protocol version was bumped; all zrepl daemons in a setup must be updated and restarted before replication can resume.

## New Users

We provide [quick-start guides](https://zrepl.github.io/v0.6.0/quickstart.html) for different usage scenarios.
Also, we recommend studying the [overview section of the configuration chapter](https://zrepl.github.io/v0.6.0/configuration/overview.html) to understand how zrepl replication works.

## Testing & Upgrading

* Read the [Changelog](https://zrepl.github.io/v0.6.0/changelog.html)
* Download & run the `platformtest` binary on a test system.
  ```
  sudo ./platformtest-linux-amd64 \
    -imagepath /tmp/zreplplatformtest.img \
    -mountpoint /tmp/zreplplatformtest \
    -poolname zreplplatformtest
  ```
* Deploy the binary release / install the package.

## Donations

zrepl is a spare-time project but nonetheless requires significant development and maintenance effort.
Please consider donating to support future development. Thank you! <br/>

[![Support me on Patreon](https://img.shields.io/badge/dynamic/json?color=yellow&label=Patreon&query=data.attributes.patron_count&suffix=%20patrons&url=https%3A%2F%2Fwww.patreon.com%2Fapi%2Fcampaigns%2F3095079)](https://patreon.com/zrepl) [![Donate via GitHub Sponsors](https://img.shields.io/static/v1?label=Sponsor&message=%E2%9D%A4&logo=GitHub&style=flat&color=yellow)](https://github.com/sponsors/problame) [![Donate via Liberapay](https://img.shields.io/liberapay/patrons/zrepl.svg?logo=liberapay)](https://liberapay.com/zrepl/donate) [![Donate via PayPal](https://img.shields.io/badge/donate-paypal-yellow.svg)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=R5QSXJVYHGX96)

## Binary Releases

The released binaries were produced with the following docker container:

```
sudo docker tag zrepl_release problame/zrepl_build:v0.6.0
sudo docker push problame/zrepl_build:v0.6.0
v0.6.0: digest: sha256:b14873f2bb754489cc54ae07b3fc1fcfaf9284f24269c2f92008cf81d39e17ea size: 4101
```

<hr /><em>This discussion was created from the release <a href='https://github.com/zrepl/zrepl/releases/tag/v0.6.0'>v0.6.0</a>.</em>

**Comment by ahjohannessen** [comment]
@problame Are binaries on their way to https://zrepl.cschwarz.com/apt ? :)

**Reply by problame** [reply]
I will update the APT and RPM mirrors in a few days to increase the chance that people learn about the release.
You can use the .deb files on GitHub in the meantime. I'll use the exact same files when publishing to the mirrors.

**Reply by ahjohannessen** [reply]
Got it. Thank you :)
